### PR TITLE
[PORT] Ports Mech Drill fixes from Occulus-Eris, Fixes/Buffs the Mech plasma cutter

### PR DESCRIPTION
**Pull request by githubuser4141**
## About The Pull Request

Simply ports a fix for mech drills breaking in one hit from Occulus-Eris, all credit goes to [This PR](https://github.com/Occulus-Server/Occulus-Eris/pull/175) for the drill fix. Credit to [Baystation](https://github.com/Baystation12/Baystation12/pulls) for providing the plasma cutter sprites.

Balance-wise, this changes mech plasma cutters to fire a (25 dam, 10 AP beam) removes uranium from the plasma cutter recipe and removes the 33% chance for the hydraulic clamp to miss its target as it's already weaker then a circular saw, even with super-heavy mech arms. Why make it even worse?

Testing-wise, I tested the drill and cutter a few times in-game and they worked, however, I messed up pushing this to Github so something might've broken inbetween or as I'm not a coder I might've missed something- so exercise caution.

## Why It's Good For The Game

Currently, mechs are useless for mining or breaking and entry as neither the drill or the plasma cutter work. This hopefully fixes that, and makes mechs more useful.

## Changelog
:cl:
add: Adds a function plasma cutter for Exosuits, a hitscan 'laser' that does 25 brute damage w/ 10 AP. Isn't a true laser and gets blocked by windows.
balance: See above/below, also changed the plasma cutter recipe to require plasteel instead of uranium. Removes the 33% chance for the Mech Clamp's melee to miss completely.
fix: Fixes the Exosuit drills having 2 durability and fixes the Exosuit plasma cutter not firing. Credit to Occulus-Eris for the fixes!
soundadd: (I think) Changes/'Adds' a new noise for the exo-drill.
imageadd: Ports Baystation plasma cutter beam sprites
code: Adds a plasma cutter for exosuits, buffs it to be slightly competitive with Eris weapons, fixes Exosuit drills breaking instantly on use
/:cl:

### CODE REVIEW

**Comment by TheShown911** [comment]
Please update the changelog.

**Comment by githubuser4141** [comment]
> Please update the changelog.

Done, I think.

File: code/modules/mechs/equipment/utility.dm [hunk 1]
```diff
@@ -205,13 +205,27 @@
...
 	..(newloc,MATERIAL_PLASTEEL)
+	src.ApplyDurability()	// OCCULUS EDIT -- apply durability to drills properly
 
 /obj/item/material/drill_head/diamond/New(var/newloc)
 	..(newloc,MATERIAL_DIAMOND)
+	src.ApplyDurability()	// OCCULUS EDIT -- apply durability to drills properly
+
+///// OCCULUS EDIT -- handy verb to prevent copy/pasting durability in each New proc
+/obj/item/material/drill_head/verb/ApplyDurability()
+	durability = 2 * (material ? material.integrity : 1)
```
Comment (TheShown911) [inline code/modules/mechs/equipment/utility.dm:228]: Doesn't need occulus edit as it's on eris now, you can also remove the `.src` from code.

File: code/modules/projectiles/effects.dm [hunk 2]
```diff
@@ -160,4 +160,16 @@
...
+// Cutter
+//----------------------------
+/obj/effect/projectile/laser/plasmacutter/tracer
+	icon_state = "plasmacutter"
+
+/obj/effect/projectile/laser/plasmacutter/impact
+	icon_state = "impact_plasmacutter"
+
+/obj/effect/projectile/laser/plasmacutter/muzzle
+	icon_state = "muzzle_plasmacutter"
```
Comment (TheShown911) [inline code/modules/projectiles/effects.dm:175]: Add an empty line at the end of the file.

File: code/modules/mechs/equipment/utility.dm (hunk 1 above)
Comment (githubuser4141) [reply code/modules/mechs/equipment/utility.dm]: got rid of the occulus stuff, what'dya mean the .src's? I am not a coder

File: code/modules/projectiles/effects.dm (hunk 2 above)
Comment (githubuser4141) [reply code/modules/projectiles/effects.dm]: Done

File: code/modules/mechs/equipment/utility.dm (hunk 1 above)
Comment (githubuser4141) [reply code/modules/mechs/equipment/utility.dm]: Got rid of the src's on the areas you mentioned

**Review by drexample — APPROVED** [review]

**Pull request merged**